pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -611,3 +470,209 @@ func (n *nodeLocalDNS) getAddress(useIPv6Brackets bool) string {
 	}
 	return nodelocaldnsconstants.IPVSIPv6Address
 }
+
+// createCleanupConfigMap creates a ConfigMap containing the cleanup shell script for node-local-dns cleanup DaemonSet.
+func (n *nodeLocalDNS) createCleanupConfigMap(ctx context.Context) error {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cleanupConfigMapName,
+			Namespace: metav1.NamespaceSystem,
+		},
+	}
+
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, n.values.ShootClient, cm, func() error {
+		cm.Data = map[string]string{
+			dataKeyCleanupScript: cleanupScript,
+		}
+		cm.Immutable = ptr.To(true)
+		return nil
+	})
+	return err
+}
+
+func (n *nodeLocalDNS) createCleanupDaemonSet(ctx context.Context) error {
+	cleanupDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      labelValueAndCleanupName,
+			Namespace: metav1.NamespaceSystem,
+			Labels: map[string]string{
+				v1beta1constants.LabelApp: labelValueAndCleanupName,
+			},
```

Comment:
Labels should also be set within the func below (consider using `metav1.SetMetaDataLabel`)